File: compute_science/binarySearch.js
function binarySearch(arr, val) {
  let left = 0;
  let right = arr.length;

  while (left <= right) {
    let middle = Math.floor((left + right) / 2);
    if (arr[middle] === val) return middle;
    if (val < arr[middle]) {
      right = middle - 1;
    } else {
      left = middle + 1;
    }
  }

  return -1;
}

console.log(binarySearch([1, 2, 3, 4, 5], 5));

File: compute_science/binary_search.py
import math


def binary_search(arr, num):
    left = 0
    right = len(arr)
    while left < right:
        mid_index = math.floor((left + right) / 2)
        if num == arr[mid_index]:
            return mid_index
        if num < arr[mid_index]:
            right = mid_index - 1
        else:
            left = mid_index + 1

    return -1


testArr = [1, 2, 5, 6, 7, 10, 100, 144, 49009]
print(binary_search(testArr, 144))
